plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AbstractFilesTable.java

```diff
@@ -0,0 +1,194 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slices;
+import io.trino.plugin.iceberg.util.PageListBuilder;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ColumnMetadata;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.SystemTable;
+import io.trino.spi.type.ArrayType;
+import io.trino.spi.type.TypeManager;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataTask;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TypeSignature.mapType;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public abstract class AbstractFilesTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    public AbstractFilesTable(SchemaTableName tableName, TypeManager typeManager, Table icebergTable)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+        this.tableMetadata = tableMetadata(tableName, typeManager);
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    protected List<Page> buildPages(Table icebergTable)
+    {
+        PageListBuilder pagesBuilder = PageListBuilder.forTable(getTableMetadata());
+        Map<Integer, Type> idToTypeMapping = getIcebergIdToTypeMapping(icebergTable.schema());
+        TableScan tableScan = buildTableScan();
+
+        tableScan.planFiles().forEach(fileScanTask -> {
+            DataTask dataTask = (DataTask) fileScanTask;
+
+            dataTask.rows().forEach(structLike -> addRow(pagesBuilder, idToTypeMapping, structLike));
+        });
+
+        return pagesBuilder.build();
+    }
+
+    protected abstract TableScan buildTableScan();
+
+    protected Table getIcebergTable()
```

Comment:
I'd remove this getter and instead pass `icebergTable` as a parameter to `buildTableScan`
```suggestion
```

Reply:
@alexjo2144 
need `icebergTable` from the Files and AllFiles, so we need to keep this method.

https://github.com/trinodb/trino/blob/b26bee561601e7423eec6b24f301b9f6666bf5f4/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java#L46